### PR TITLE
major: added v3 plugin for env-api

### PR DESCRIPTION
**Pull request by c1freitas**
# What 
Add a v3 env-api plugin. This required changes to the plugin interface so bumping the version. 

This plugin supports falling back to the v2 endpoint.

### CODE REVIEW

File: test/unit/plugin/env-api-client-v3.spec.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+				envs = yield apiConfig.fetch(testService, clusterDef);
+				expect(rp.callCount).to.equal(1);
+
+				expect(envs.env.length).to.equal(5);
+				expect(envs.env[0].name).to.equal("GET_HOSTS_FROM");
+				expect(envs.env[0].value).to.equal("dns");
+				expect(envs.env[1].name).to.equal("MAX_RETRIES");
+				expect(envs.env[1].value).to.equal("0");
+
+				// console.log("\n V3 result" + JSON.stringify(envs))
```
Comment (chesleybrown) [inline test/unit/plugin/env-api-client-v3.spec.js]: Some debug comments left in here.

File: src/lib/cluster-definition.js [hunk 2]
```diff
@@ -31,6 +31,13 @@ class ClusterDefinition {
 		return this.cluster.metadata.name;
 	}
 
+  /**
+   * Helper method for accessing metadata
+   */
```
Comment (chesleybrown) [inline src/lib/cluster-definition.js]: Odd indent here.

File: src/lib/cluster-definition.js [hunk 3]
```diff
@@ -31,6 +31,13 @@ class ClusterDefinition {
 		return this.cluster.metadata.name;
 	}
 
+  /**
```
Comment (chesleybrown) [inline src/lib/cluster-definition.js]: Odd indent here.

Comment (c1freitas) [inline src/lib/cluster-definition.js]: where is `gofmt` when you need it?

File: test/unit/plugin/env-api-client-v3.spec.js (hunk 1 above)
Comment (c1freitas) [inline test/unit/plugin/env-api-client-v3.spec.js]: fixed

**Pull request merged**